external/upstream/gauxc/CMakeLists.txt

```diff
@@ -0,0 +1,91 @@
+if(${ENABLE_gauxc})
+    if(NOT "${gauxc_DIR}" STREQUAL "")
+        if("${IntegratorXX_DIR}" STREQUAL "")
+            set(IntegratorXX_DIR ${gauxc_DIR}/../IntegratorXX)
+        endif()
+        set(ExchCXX_DIR ${gauxc_DIR}/../ExchCXX)
+    endif()
+
+    get_property(BLAS_LIBRARIES TARGET lapk PROPERTY INTERFACE_LINK_LIBRARIES)
+
+    find_package(gauxc CONFIG) 
+
+    if(${gauxc_FOUND})
+        #== check if GPU-enabled GauXC is found, if GPU is specified ==#
+        get_property(_gpu TARGET gauxc::gauxc PROPERTY GAUXC_HAS_DEVICE)
+        
+        if(${gauxc_ENABLE_GPU} AND (NOT ${_gpu}))
+           message(FATAL_ERROR "gauxc_ENABLE_GPU turned on, but selected GauXC install is not built for GPU support!")
+        elseif((NOT ${gauxc_ENABLE_GPU}) AND ${_gpu})
+           message(FATAL_ERROR "gauxc_ENABLE_GPU turned off, but selected GauXC install is built for GPU support!")
+        endif()
+
+        #== check if MAGMA-enabled GauXC is found, if MAGMA is specified ==#
+        get_property(_magma TARGET gauxc::gauxc PROPERTY GAUXC_HAS_MAGMA)
+        
+        if(${gauxc_ENABLE_MAGMA} AND (NOT ${GAUXC_ENABLE_GPU}))
+           message(FATAL_ERROR "gauxc_ENABLE_MAGMA turned on, but gauxc_ENABLE_GPU is turned off! Magma only works with device-enabled GauXC instances.")
+        endif()
+        
+        if(${gauxc_ENABLE_MAGMA} AND (NOT ${_magma}))
+           message(FATAL_ERROR "gauxc_ENABLE_MAGMA turned on, but selected GauXC install is not built for MAGMA support!")
+        elseif((NOT ${gauxc_ENABLE_MAGMA}) AND ${_magma})
+           message(FATAL_ERROR "gauxc_ENABLE_MAGMA turned off, but selected GauXC install is built for MAGMA support!")
+        endif()
+
+        #== normal print-out and library addition ==#
+        get_property(_loc TARGET gauxc::gauxc PROPERTY IMPORTED_LOCATION_RELWITHDEBINFO)
+        #get_property(_loc TARGET gauxc::gauxc PROPERTY IMPORTED_LOCATION)
+        message(STATUS "${Cyan}Found gauxc ${ColourReset}: ${_loc} (found version ${gauxc_VERSION})")
+        add_library(gauxc_external INTERFACE)  # dummy
+
+        if(${CMAKE_INSIST_FIND_PACKAGE_gauxc})
+            message(VERBOSE "Suitable gauxc located externally as user insists")
+        endif()
+
+    else()
+        if(${CMAKE_INSIST_FIND_PACKAGE_gauxc})
+            message(FATAL_ERROR "Suitable gauxc could not be externally located as user insists")
+        endif()
+
+        if(${gauxc_ENABLE_GPU} AND (NOT DEFINED ${CMAKE_CUDA_ARCHITECTURES}))
+           message(FATAL_ERROR "Internal GPU-enabled GauXC construction specified, but CMAKE_CUDA_ARCHITECTURES is not defined!")
+        endif()
+
+        message(STATUS "Suitable gauxc could not be located, ${Magenta}Building gauxc${ColourReset} instead.")
+
+        include(ExternalProject)
+        ExternalProject_Add(gauxc_external
+            DEPENDS lapack_external 
+                    libxc_external
+                    gau2grid_external
```

Comment:
IntegratorXX isn't needed in the externalproject_add()?

Reply:
Technically no, because GauXC will just build an IntegratorXX instance along itself here. 

Right now, the GauXC build system, whether hooking up to an external GauXC install or building GauXC internally, is set to only ever use internally-built instances (as in, built internally in GauXC) of ExchCXX and IntegratorXX for those dependencies. The question is, is this desirable behavior? I know @wavefunction91 mentioned this being desirable for ExchCXX, but I'm not sure if that carries over to IntegratorXX as well.